### DISCUSSION TITLE
Couldn't run binary after using $GITHUB_PATH

### DISCUSSION THREAD

**Discussion by michalbarer1** [community/community#26373] [Actions]
<p>Hi Github community,<br>
I have trouble executing a binary I download since I stopped using the deprecated <code>add-path</code> command.</p>
<p>This worked for me:</p>
<pre><code class="lang-auto">      - name: Add workspace to path
        if: startsWith(runner.os, 'Windows')
        run: |
          echo "::add-path::${env:GITHUB_WORKSPACE}"
      - name: Install minio client
        if: startsWith(runner.os, 'Windows')
        run: |
          curl "https://dl.min.io/client/mc/release/windows-amd64/archive/mc.RELEASE.2020-08-08T02-33-58Z" --output mc.exe
</code></pre>
<p>now I switched to this:</p>
<pre><code class="lang-auto">      - name: Add workspace to path
        if: startsWith(runner.os, 'Windows')
        run: |
          echo "${GITHUB_WORKSPACE}" &gt;&gt; $GITHUB_PATH

      - name: Install minio client
        if: startsWith(runner.os, 'Windows')
        run: |
          curl "https://dl.min.io/client/mc/release/windows-amd64/archive/mc.RELEASE.2020-08-08T02-33-58Z" --output mc.exe
</code></pre>
<p>and the <code>mc</code> command isn’t recognized.</p>
<p>Thank you for your help!</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>That’s probably because you’re trying to use bash syntax on Windows without setting the shell to bash. See this thread for a similar problem, the solutions described there should work for you, too:</p><aside class="quote quote-modified" data-post="1" data-topic="143992">
  <div class="title">
    <div class="quote-controls"></div>
    
    <a href="https://github.community/t/github-path-does-not-add-to-the-path/143992">$GITHUB_PATH does not add to the path</a> <a class="badge-wrapper  bullet" href="/c/code-to-cloud/github-actions/41"><span class="badge-category-parent-bg" style="background-color: #6a737d;"></span><span class="badge-category-bg" style="background-color: #2188ff;"></span><span style="" data-drop-close="true" class="badge-category clear-badge" title="Discussion and questions about GitHub Actions—automation for all of your development workflows, now with world-class CI/CD.">GitHub Actions</span></a>
  </div>
  <blockquote>
    Hi Github Community, 
I’m having trouble using $GITHUB_PATH. I can’t tell if I’m not using it correctly, or if It’s not executing correctly. 
For both of the following links, please pay attention to the Windows builds only. 
<a href="https://github.com/jameskr97/etterna/runs/1407721809" rel="noopener nofollow ugc">Here is a link</a> to a  previous version of a Github Action script that did work prior the switch to $GITHUB_PATH. 
<a href="https://github.com/jameskr97/etterna/runs/1414441535" rel="noopener nofollow ugc">Here is a link</a> to the most recent version of the Github Action script that stopped working with the $GITHUB_PATH changes. 
Please note specifically the steps titl…
  </blockquote>
</aside>

**Comment by michalbarer1** [comment]
<p><a class="mention" href="/airtower-luna">@airtower-luna</a> thanks! It helped. Is this new?  I never specified shell before, and everything was running well.</p>

**Comment by airtower-luna** [comment]
<p>I guess with the old add-path syntax the difference was small enough that in most cases it just worked. I mean, in <code>echo "::add-path::${env:GITHUB_WORKSPACE}"</code> the only difference is the <code>env:</code> prefix. :wink:</p>
